### PR TITLE
Increase Utility coverage

### PR DESCRIPTION
**Pull request by jennuine**
Part of https://github.com/gazebosim/gz-sim/issues/1575

## Summary
- Increased `Utility.cc` coverage by adding tests for `ConvertFuelMetadata` methods
- Updated `ConvertFuelMetadata(string, msgs::FuelMetadata)` function to take care of `<world>`: a1d294d585853553e5d89e32c61d3716950f73f2 (commit message is backwards :sweat_smile:)
- Removed unused `split` function: 8ff8356a98ba47c1206e466f074e68e46e059110

## Test it
```bash
./path/to/build/UNIT_Utility_TEST --gtest_filter="*ConvertFuelMetadata"
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/Utility.cc
Comment (chapulina) [inline src/Utility.cc:973]: Nitpick that this isn't a "model" element anymore. The variable could be updated to be more generic, and the comment in line 967 could mention `<world>` as well.

File: src/Utility.cc
Comment (chapulina) [inline src/Utility.cc:1108]: The `<sdf>` element should go outside the `<model>` / `<world>`

**Review by chapulina — CHANGES_REQUESTED** [review]
Great to see the coverage efforts resulting in implementation improvements 😃

File: src/Utility.cc
Comment (jennuine) [reply src/Utility.cc:1108]: It does for `*.sdf` files but not `*.config` files. Here are some examples:
- https://github.com/gazebosim/sdformat/blob/sdf12/test/integration/model/box/model.config#L5
- https://github.com/gazebosim/sdformat/blob/sdf12/test/integration/model/robot/model.config#L4

Comment (chapulina) [reply src/Utility.cc:1108]: Ah you're right, I got them mixed up 🙃

**Review by chapulina — APPROVED** [review]
LGTM, I'll leave it up to you if you think the nitpick is worth addressing.

File: src/Utility.cc
Comment (jennuine) [reply src/Utility.cc]: Done c86f6d101a65e5dd7ac18664fe5da3f78bd71a1a :smile:

**Pull request merged**
